gray/rtl/BUILD.bazel

```diff
@@ -0,0 +1,50 @@
+# Copyright 2024 The Bedrock-RTL Authors
```

Comment:
Please move this under the `enc` group which is generally containing simple combinational modules. Especially those that do invertible encodings (like the existing `br_enc_bin2onehot` and `br_enc_onehot2bin`.

Suggest this scheme to stay consistent:
`enc/rtl/br_enc_bin2gray`
`enc/rtl/br_enc_gray2bin`

Reply:
Done.